module-4/backend/week_two.md

```diff
@@ -5,18 +5,36 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
+>* function keyword in ES5 vs fat arrows in ES6 
+2. What's the difference between asynchronous and synchronous JavaScript?
+>* asynchronous means that multiple js processes can be executed at different times while synchronous is that they are performed at the same time.  
 3. What are the four pillars of Object Oriented programming?
+>* APEI => Abstraction, Polymorphism, Encapsulation, Inheritance 
 4. What are some tools available in JavaScript to help you write object oriented code?
+>* Use class syntax with constructor method like initialize in Ruby and new Object declarations to help write OOP style.
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+>* the meaning of this is different between ES5 and Es6 so this is something to be aware of. 
 6. What's a callback function and what are some reasons when we use/need callback functions?
+>* a callback function is when we invoke the function with parenthesis myfuntion() to be executed in the call stack. 
```

Comment:
a callback function is a function that's passed as an argument to another function and only executes upon the initial one firing